source/slang/slang-ir-autodiff-rev.cpp

```diff
@@ -568,7 +569,15 @@ namespace Slang
             tempBuilder.setInsertBefore(diffPropagateFunc);
         }
 
-        auto fwdDiffFunc = generateNewForwardDerivativeForFunc(&tempBuilder, primalFunc, diffPropagateFunc);
+        // Normalize the CFG of the primal func before starting the reverse-mode pass.
+        IRCloneEnv cfgNormCloneEnv;
```

Comment:
After merging with top of tree, you will see that `generateNewForwardDerivativeFunc` calls `prepareForBackwardAutoDiff` which makes a clone of the primal func and then makes some preparational changes to the clone of primal func before applying forward diff to it, including single-return transform, multi-level-break elimination, and the array/struct type SSA transform. The `normalizeCFG` step should be inserted there instead, and we need to make sure it is called before the array/struct type SSA step to avoid undoing the SSA transforms since normalizeCFG calls `elminatePhi` to get out of SSA first.